### PR TITLE
Go 1.19 compat

### PR DESCRIPTION
**Pull request by Didainius**
As part of Go 1.19 rollout this PR adjusts a few things:
* Bumps Go to 1.18 in `go.mod` as the minimum required version.
* Does `make fmt` which adjusts doc formatting for Go 1.19
* Adjusts GitHub actions to use the latest code
* `staticcheck` switched version naming from `2021.1.2` to `v0.3.3` in downloads section. This PR also updates the code to fetch correct staticcheck.
* package `io/ioutil` is deprecated as of Go 1.16. `staticcheck` started complaining about usage of deprecated packages. As a result this PR switches packages to either `io` or `os` (still the same functions are used)

### CODE REVIEW

File: .changes/v3.8.0/902-notes.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+* Bumps Go to 1.18 in `go.mod` as the minimum required version. [GH-902]
```
Comment (lvirbalas) [inline .changes/v3.8.0/902-notes.md:1]:
```suggestion
* Bump Go to 1.18 in `go.mod` as the minimum required version. [GH-902]
```

File: .changes/v3.8.0/902-notes.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+* Does `make fmt` which adjusts doc formatting for Go 1.19 [GH-902]
```
Comment (lvirbalas) [inline .changes/v3.8.0/902-notes.md:2]: What/who does fmt? :)

File: .changes/v3.8.0/902-notes.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+* Adjusts GitHub actions to use the latest code [GH-902]
```
Comment (lvirbalas) [inline .changes/v3.8.0/902-notes.md:3]:
```suggestion
* Adjust GitHub actions to use the latest code [GH-902]
```
But this raises an interesting question: how to procede with changelog entries which are not very actual to end-users. But we can amend them during pre-release changelog review.

**Review by lvirbalas — APPROVED** [review]
Some asks, and then LGTM  :)

File: .changes/v3.8.0/902-notes.md (hunk 1 above)
Comment (Didainius) [reply .changes/v3.8.0/902-notes.md]: Fixed.

File: .changes/v3.8.0/902-notes.md (hunk 2 above)
Comment (Didainius) [reply .changes/v3.8.0/902-notes.md]: Rephrased

File: .changes/v3.8.0/902-notes.md (hunk 3 above)
Comment (Didainius) [reply .changes/v3.8.0/902-notes.md]: Adjusted. Well, I am putting it in `notes` section which is probably the best we can get. In the end we might skip it at all during final release if we decide to.

**Pull request merged**